src/db.rs

```diff
@@ -637,25 +638,27 @@ impl<T: ThreadMode> DBWithThreadMode<T> {
                     cfhandles.as_mut_ptr(),
                 )),
                 AccessType::Secondary { secondary_path } => {
+                    let secondary_cpath = to_cpath(secondary_path)?;
                     ffi_try!(ffi::rocksdb_open_as_secondary_column_families(
                         opts.inner,
                         cpath.as_ptr() as *const _,
-                        to_cpath(secondary_path)?.as_ptr() as *const _,
+                        secondary_cpath.as_ptr() as *const _,
                         cfs_v.len() as c_int,
                         cfnames.as_ptr(),
                         cfopts.as_ptr(),
                         cfhandles.as_mut_ptr(),
                     ))
                 }
                 AccessType::WithTTL { ttl } => {
+                    let ttls_v = vec![ttl.as_secs() as c_int; cfs_v.len()];
                     ffi_try!(ffi::rocksdb_open_column_families_with_ttl(
                         opts.inner,
                         cpath.as_ptr(),
                         cfs_v.len() as c_int,
                         cfnames.as_ptr(),
                         cfopts.as_ptr(),
                         cfhandles.as_mut_ptr(),
-                        &(ttl.as_secs() as c_int) as *const _,
+                        ttls_v.as_ptr(),
```

Comment:
I believe this was undefined behavior because the temporary `c_int` would be out of scope before the pointer is used in the called function.

Aditionally there would be an out-of-bounds read if `cfs_v.len() > 1`, because the implementation expects a pointer to `cfs_v.len()` consecutive entries.

Reply:
Perhaps I'm missing something, but where can out-of-bound read occur? `cfnames`? But it is an array, so everything should be OK. Am I missing something?

Reply:
I mean the `ttls` array. Previously we were passing a pointer to a single integer value, but `n` integer values are expected (https://github.com/facebook/rocksdb/blob/f20b674796ffd7ca32471705876fc651b8e246db/db/c.cc#L827-L828), one for each column family.

Reply:
I mean the `ttls` array. Previously we were passing a pointer to a single integer value, but `n` integer values are expected (https://github.com/facebook/rocksdb/blob/f20b674796ffd7ca32471705876fc651b8e246db/db/c.cc#L827-L828), one for each column family.